### PR TITLE
Corrections to HTML5 video ICE message cycle and the share video button

### PR DESCRIPTION
**Pull request by lfzawacki**

### CODE REVIEW

File: bigbluebutton-html5/imports/ui/components/video-dock/component.jsx [hunk 1]
```diff
@@ -105,7 +105,11 @@ class VideoDock extends Component {
 
     for (let i = 0; i < users.length; i++) {
       if (users[i].has_stream && users[i].userId !== userId) {
-        this.start(users[i].userId, false);
+        // FIX: Really ugly hack, but sometimes the ICE candidates aren't
+        // generated properly when we send videos right after componentDidMount
+        setTimeout(() => {
+          this.start(users[i].userId, false);
+        }, 2000);
```
Comment (OZhurbenko) [inline bigbluebutton-html5/imports/ui/components/video-dock/component.jsx]: Setting timeout to 0 doesn't help? only 2000 fixes it for you?
(if that's the case, then we'll see this issue again)

**Comment by prlanzarin** [comment]
Regarding the `VideoDock` component doing too much stuff: yup, we're aware of it and we agree. It's in our plans to move some of the stuff out. The intention is to have a service/API that communicates with `bbb-webrtc-sfu` and can be reused anywhere. Some work was done on that intent (basically unifying some of the component logic with the `kurento-extension` script), but we had to stop it to focus on the stability issues. We hope we'll get back to it soon.

**Comment by oswaldoacauan** [comment]
@prlanzarin sure, I didn't expect this PR to solve that anyway. Left a comment just so we are aware of the issue :+1:

**Comment by Tainan404** [comment]
What should the correction to ICE message cycles fix? I couldn`t test this part of the PR since I dont actually know what was/should be fixed.

Could you explain a little further what does that mean and what case it should fix so I can actually test it.

**Comment by prlanzarin** [comment]
@Tainan404 the test case is basically related to Firefox late joiners as viewers. Try sharing your webcam on a couple of Chrome instances, wait a few moments and then join with firefox in the same room. You should be seeing the Chrome webcams.

**Comment by prlanzarin** [comment]
@Tainan404 "When I close the server bbb-webrtc-sfu the icon should not return to normal or show a notification?"

This is a server-side mishap and I'm fixing it on my next PR.

File: labs/bbb-webrtc-sfu/lib/video/VideoManager.js [hunk 2]
```diff
@@ -31,35 +31,34 @@ var _onMessage = function (_message) {
...
-      }
-      else if (message.cameraId && sessions[sessionId][message.cameraId]) {
-        video = sessions[sessionId][message.cameraId];
-      }
+  if (!iceQueues[sessionId]) {
+      iceQueues[sessionId] = {};
+  }
 
-      break;
+  if (typeof sessions[sessionId][cameraId] !== 'undefined' &&  sessions[sessionId][cameraId]) {
```
[the remark this answers is not in the document]
Comment (antobinary) [reply labs/bbb-webrtc-sfu/lib/video/VideoManager.js]: So: `if (sessions[sessionId][cameraId]) {...` ?

[the remark this answers is not in the document]
Comment (Tainan404) [reply labs/bbb-webrtc-sfu/lib/video/VideoManager.js]: Yes.

**Comment by lfzawacki** [comment]
Made some more changes. Paulo is working on a lot of small changes and corrections in the bbb-webrtc-sfu component and it's probably better to worry about the rest of the style issues in future PRs.

**Pull request merged**